cpp/src/gandiva/configuration.h

```diff
@@ -35,6 +35,8 @@ class GANDIVA_EXPORT Configuration {
  public:
   friend class ConfigurationBuilder;
 
+  explicit Configuration(bool optimize) : optimize_(optimize) {}
```

Comment:
also have a default flag i guess (and remove the default below)..to make it consistent for other users (if any)